airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/relationaldb/state/CursorManager.kt

```diff
@@ -103,7 +102,7 @@ class CursorManager<S : Any>(
                 .map { stream: AirbyteStream ->
                     AirbyteStreamNameNamespacePair.fromAirbyteStream(stream)
                 }
-                .collect(Collectors.toSet())
+                .toMutableSet()
```

Comment:
why this specific one is a mutable set ? did some test catch that it needs to be mutated

Reply:
the line below does an addAll